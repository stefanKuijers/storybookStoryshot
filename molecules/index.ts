export { default as Avatar } from "./Avatar/Avatar";

export { default as Dropdown } from "./Dropdown/Dropdown";
export { default as DropdownItem } from "./DropdownItem/DropdownItem";
export { default as DropdownItemNavLink } from "./DropdownItemNavLink/DropdownItemNavLink";
