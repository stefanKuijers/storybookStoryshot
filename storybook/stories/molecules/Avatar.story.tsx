import { text } from "@storybook/addon-knobs";
import React from "react";

import { Avatar } from "../../../molecules";
import { stageStoriesOf } from "../utils";

stageStoriesOf("Molecules/Avatar", "Avatar")
  .add("Simple Avatar", () => {
    return (
      <>
        <Avatar label={text("label", "DL")} />
        <Avatar name={text("User's name", "John Doe")} />
      </>
    );
  })
  .add("Avatar with info", () => {
    return (
      <>
        <Avatar
          name={text("User' name", "John Doe")}
          userName={text("Username", "john88")}
          emailAddress={text("Email address", "john88@mail.com")}
        />
      </>
    );
  });
