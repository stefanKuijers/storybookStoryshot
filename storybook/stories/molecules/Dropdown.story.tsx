import { boolean, text } from "@storybook/addon-knobs";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { Placement } from "reactstrap/lib/Popper";

import { Title } from "../../../atoms";
import { Dropdown, DropdownItem, DropdownItemNavLink } from "../../../molecules";
import { stageStoriesOf, t } from "../utils";

const dropdownItems = [
  {
    children: "Item 1",
    type: DropdownItem,
    options: {}
  },
  {
    children: "Item 2",
    type: DropdownItem,
    options: {}
  },
  {
    children: "Item 3",
    type: DropdownItem,
    options: {}
  },
  {
    children: "Item 4",
    type: DropdownItem,
    options: {}
  }
];

const headerMenuDropdownItems = [
  {
    type: DropdownItemNavLink,
    children: "Active Projects (13)",
    options: {
      to: "/projects/active",
      icon: "active"
    }
  },
  {
    type: DropdownItemNavLink,
    children: "Removed Projects (1)",
    options: {
      to: "/projects/removed",
      icon: "trash"
    }
  },
  {
    type: DropdownItemNavLink,
    children: "Archived Projects (432)",
    options: {
      to: "/projects/archived",
      icon: "archived"
    }
  }
];

const placement: Placement = "auto";

stageStoriesOf("Molecules/Dropdown", "Dropdown")
  .add("Dropdown", () => (
    <Dropdown
      id="DropdownDefaultId"
      content={text("Custom child", "Lorem ipsum")}
      dropdownItems={dropdownItems}
      placement={placement}
      containerClassName={text("Container class name?", "")}
      className={text("Custom class?", "")}
      showArrow={boolean("Show arrow?", false)}
    />
  ))
  .add("Header Dropdown", () => (
    <Router>
      <Dropdown
        id={"headerDropdown"}
        content={<Title className={"HeaderTitle"} type={"h1"} title={t("This stuff")} />}
        dropdownItems={headerMenuDropdownItems}
        showArrow
      />
    </Router>
  ));
