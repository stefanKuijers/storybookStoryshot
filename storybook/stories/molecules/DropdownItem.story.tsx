import { boolean, text } from "@storybook/addon-knobs";
import React from "react";

import { DropdownItem } from "../../../molecules";
import { stageStoriesOf } from "../utils";

stageStoriesOf("Molecules/Dropdown", "DropdownItem").add("DropdownItem", () => (
  <DropdownItem isSeparator={boolean("Is separator?", false)} icon={"fire"} onClick={console.log}>
    {text("Custom child?", "Lorem ipsum")}
  </DropdownItem>
));
