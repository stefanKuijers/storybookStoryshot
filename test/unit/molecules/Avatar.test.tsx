import React from "react";
import renderer from "react-test-renderer";

import { Avatar } from "../../../molecules";

describe("Avatar", () => {
  describe("with property name", () => {
    const Renderer = renderer.create(<Avatar name={"John Doe"} />);
    const Component = Renderer.root;

    it("matches the snapshot", () => {
      expect(Renderer.toJSON()).toMatchSnapshot();
    });

    it("should contain abreviated initials", () => {
      expect(Component.findByProps({ className: "Avatar__label" }).children).toEqual(["JD"]);
    });
  });

  describe("with additional information", () => {
    const Renderer = renderer.create(
      <Avatar name={"John Doe"} userName={"john88"} emailAddress={"john88@mail.com"} />
    );
    const Component = Renderer.root;

    it("matches the snapshot", () => {
      expect(Renderer.toJSON()).toMatchSnapshot();
    });

    it("should contain a popover rendered by DropDown", () => {
      expect(Component.findByProps({ className: "AvatarDropdown" }).children.length).toEqual(1);
    });
  });
});
