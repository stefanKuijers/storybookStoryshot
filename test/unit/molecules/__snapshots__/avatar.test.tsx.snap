// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar with additional information matches the snapshot 1`] = `
<div
  className="Dropdown"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    className="Dropdown__link"
    id="john88-info"
    onClick={[Function]}
  >
    <span
      className="Dropdown__title"
    >
      <div
        className="Avatar tclass-avatar"
      >
        <span
          className="Avatar__label"
        >
          JD
        </span>
      </div>
    </span>
  </a>
</div>
`;

exports[`Avatar with property name matches the snapshot 1`] = `
<div
  className="Avatar tclass-avatar"
>
  <span
    className="Avatar__label"
  >
    JD
  </span>
</div>
`;
